File: Maxglass.Ecommerce/Maxglass.Ecommerce.Dominio/Utils/Consulta/PaginacaoConsulta.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Maxglass.Ecommerce.Dominio.Utils.Consulta
{
    public class PaginacaoConsulta<T>
    {
        public int Quantidade { get; set; }
        public IList<T> Registros{get;set;}


        public PaginacaoConsulta(int quantidade, IList<T> registros){
            this.Quantidade = quantidade;
            this.Registros = registros;
        }
        
    }
}

File: Maxglass.Ecommerce.Web/src/app/produtos/models/paginacao-consulta.ts
export class PaginacaoConsulta<T> {
    quantidade: number;
    registros: Array<T>;

    constructor(params: Partial<PaginacaoConsulta<T>>) {
        this.quantidade = params.quantidade || null;
        this.registros = params.registros || null;
        
    }
}
